core/src/net/cidr/v4.rs

```diff
@@ -45,6 +64,49 @@ impl Default for Ipv4Cidr {
     }
 }
 
+#[derive(Clone, Debug)]
+pub struct Ipv4CidrIterator {
+    next: Option<u32>,
+    end: u32,
+}
+
+impl Ipv4CidrIterator {
+    fn new(start: Ipv4Addr, end: usize) -> Self {
+        let start = u32::from(start);
+        Ipv4CidrIterator {
+            next: Some(start),
+            end: start + (end as u32 - 1),
+        }
+    }
+}
+
+impl Iterator for Ipv4CidrIterator {
+    type Item = Ipv4Addr;
+
+    fn next(&mut self) -> Option<Ipv4Addr> {
+        let next = self.next?;
+        self.next = if next == self.end {
+            None
+        } else {
+            Some(next + 1)
+        };
+        Some(next.into())
+    }
+}
+
+impl IntoIterator for Ipv4Cidr {
```

Comment:
still feels redundant. `cidr.into()` and `cidr.iter()`.

Reply:
just going with `fn iter` then.